### PR TITLE
Update the FlutterLogo colors and gradients

### LINKED ISSUE #16713: Update FlutterLogoDecoration
We should change "normal" to "multiply" on this line:

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/flutter_logo.dart#L333

...and update the gradient colours to 0x661A237E at top 0 and 0x001A237E at stop 1.

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Updates the FlutterLogo colors and gradients to the latest version provided by our branding experts. As per their request, the option to customize those colors has also been removed from the FlutterLogo widget.

New look of the `FlutterLogo` widget:

material flutter_logo

## Related Issues

Fixes https://github.com/flutter/flutter/issues/16713.
https://github.com/flutter/flutter/issues/60539

## Tests

I added the following tests:

* Added screenshot tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/62052

**Comment by HansMuller** [comment]
> As per their request, the option to customize those colors has also been removed from the FlutterLogo widget.

So this is a breaking change?

File: packages/flutter/lib/src/painting/flutter_logo.dart [hunk 1]
```diff
@@ -38,53 +38,54 @@ enum FlutterLogoStyle {
...
   ///
-  /// The [lightColor], [darkColor], [textColor], [style], and [margin]
-  /// arguments must not be null.
+  /// The [lightColor], [mediumColor], [darkColor], [textColor], [style], and
+  /// [margin] arguments must not be null.
   const FlutterLogoDecoration({
-    this.lightColor = const Color(0xFF42A5F5), // Colors.blue[400]
-    this.darkColor = const Color(0xFF0D47A1), // Colors.blue[900]
-    this.textColor = const Color(0xFF616161),
+    this.lightColor = const Color(0xFF54C5F8),
```
Comment (HansMuller) [inline packages/flutter/lib/src/painting/flutter_logo.dart:48]: If we're disallowing color changes in the logo widget, why allow them here?

**Comment by goderbauer** [comment]
> So this is a breaking change?

Depends on the definition of "breaking". According to https://github.com/flutter/flutter/wiki/Tree-hygiene#1-determine-if-your-change-is-a-breaking-change this is not a breaking change.

File: packages/flutter/lib/src/painting/flutter_logo.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/painting/flutter_logo.dart:48]: No good reason. I'll remove it from here as well.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
The test is failing because of https://github.com/flutter/flutter/issues/62146.

**Pull request merged**